ingest/src/test/java/com/microsoft/azure/kusto/ingest/E2ETest.java

```diff
@@ -429,6 +435,59 @@ private boolean canAuthenticate(ConnectionStringBuilder engineCsb) {
         return true;
     }
 
+    private void assertUrlCompare(String connectionDataSource, String clusterUrl, boolean autoCorrectEndpoint, boolean isQueued) {
+        if (!autoCorrectEndpoint || clusterUrl.contains(INGEST_PREFIX) || isReservedHostname(clusterUrl)) {
```

Comment:
Don't put logic like this in tests, it makes the test useless.
Put the desired result for each url.

Reply:
WDYM?

Reply:
You are remaking the logic of what you are testing inside E2ETest.
That makes the test only check if the logic is the same in both places, and not if it is correct.
Instead, compare each url against the expected result, hardcoded.